src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -746,6 +672,7 @@ private static int WriteHeader(MessageOpcode opcode, byte[] sendBuffer, ReadOnly
                 {
                     throw new OperationCanceledException(nameof(WebSocketState.Aborted), exc);
                 }
+                _abortSource.Cancel();
```

Comment:
This is the only line of the actual fix. If we see that the receive failed but that it wasn't explicitly aborted by the user, update the status ourselves.